usb/backend/libusb0.py

```diff
@@ -619,27 +619,47 @@ def reset_device(self, dev_handle):
 
     @methodtrace(_logger)
     def is_kernel_driver_active(self, dev_handle, intf):
+        # libusb0's usb_get_driver_np() is platform specific / not portable.
+        # This function checks what kernel module handles dev_handle/intf.
+        # Various implementations may return system specific values - for
+        # instance Linux may return errno.ENODATA not present on FreeBSD.
+        # On FreeBSD successul call returns 0 otherwise -1.
+
         if not hasattr(_lib, 'usb_get_driver_np'):
             raise NotImplementedError(self.is_kernel_driver_active.__name__)
-        from errno import ENODATA
         buf = usb.util.create_buffer(_USBFS_MAXDRIVERNAME + 1)
         name, length = buf.buffer_info()
         length *= buf.itemsize
-        # based on the implementation of libusb_kernel_driver_active
+
+        # Linux: based on the implementation of libusb_kernel_driver_active
         # (see libusb/os/linux_usbfs.c @@ op_kernel_driver_active):
         # usb_get_driver_np fails with ENODATA when no kernel driver is bound,
         # and if 'usbfs' is bound that means that a userspace program is
         # controlling the device (e.g. using this very library)
+        if "linux" in sys.platform:
+            try:
+                from errno import ENODATA
+            except:
+                raise NotImplementedError(self.is_kernel_driver_active.__name__)
+
```

Comment:
This is not necessary, `errno` is already imported at the top of the module, and you're already doing `errno.ENODATA` bellow.

Reply:
Yes, this is additional check, `ENODATA` may not be defined on (some older) Linux (?), this will be explicitly inidcated with  `NotImplementedError` exception, instead throwing unhandled exception later :-)